src/commands/init.ts

```diff
@@ -133,6 +195,27 @@ export abstract class InitCommand extends BaseCommand {
           },
         ]);
         from_path = path.resolve(untildify(answers.from_compose));
+        await this.runArchitectYamlConversion(from_path, args.name, flags['component-file']);
+      }
+      return;
+    }
+
+    await this.runProjectCreation(args.name);
+  }
+
+  async getComposeFromPath(flags: any): Promise<string | undefined> {
+    let from_path;
+    if (flags['from-compose']) {
+      from_path = path.resolve(untildify(flags['from-compose']));
+    } else {
+      const files_in_current_dir = fs.readdirSync('.');
+      const default_compose = files_in_current_dir.find(f => f.includes('compose') && (f.endsWith('.yml') || f.endsWith('.yaml')));
+
+      if (default_compose) {
+        from_path = default_compose;
+        if (!fs.existsSync(from_path) || !fs.statSync(from_path).isFile()) {
+          throw new Error(`The Docker Compose file ${from_path} couldn't be found.`);
+        }
```

Comment:
Similar thing here, this is an async method but we're only doing synchronous operations, so we can make the method sync or use the async version of readdir/exists/stats to benefit from async-ness 😛

Reply:
I'm keeping this one sync as well - we need to make sure we can actually get the compose file; otherwise, it will throw undefined for the `from_path` variable and cause the file to not be found and fail the tests